workflow/doc.go

```diff
@@ -139,8 +139,79 @@ Time related functions:
+cancellation and termination.  Termination is forceful, cancellation allows a workflow to exit gracefully.
+
+Workflows can also time out, based on their ExecutionStartToClose duration.  A timeout behaves the same as
+termination (it is a hard deadline on the workflow), but a different close status and final event will be reported.
+
+Terminating a Workflow
+
+Terminating is roughly equivalent to using `kill -9` on a process - the workflow will be ended immediately,
+and no further decisions will be made.  It cannot be prevented or delayed by the workflow, or by any configuration.
+Any in-progress decisions or activities will fail whenever they next communicate with Cadence's servers, i.e. when
+they complete or when they next heartbeat.
+
+Because termination does not allow for any further code to be run, this also means your workflow has no
+chance to clean up after itself (e.g. running a cleanup Activity to adjust a database record).
+If you need to run additional logic when your workflow, use cancellation instead.
+
+Canceling a Workflow
+
+Canceling marks a workflow as canceled (this is a one-time, one-way operation), and immediately wakes the workflow
+up to process the cancellation (schedules a new decision task).  When the workflow resumes after being canceled,
+the context that was passed into the workflow (and thus all derived contexts) will be canceled, which changes the
+behavior of many workflow.* functions.
+
+Canceled workflow.Context behavior
+
+A workflow's context can be canceled by either canceling the workflow, or calling the cancel-func returned from
+a worfklow.WithCancel(ctx) call.  Both behave identically.
+
+At any time, you can convert a canceled (or could-be-canceled) context into a non-canceled context by using
+workflow.NewDisconnectedContext.  The resulting context will ignore cancellation from the context it is derived from.
+Disconnected contexts like this can be created before or after a context has been canceled, and it does not matter
+how the cancellation occurred.
+Because this context will not be canceled, this can be useful for using context cancellation as a way to request that
+some behavior be shut down, while allowing you to run cleanup logic in activities or elsewhere.
+
+As a general guideline, doing anything with I/O with a canceled context (e.g. executing an activity, starting a
+child workflow, sleeping) will fail rather than cause external changes.  Detailed descriptions are available in
+documentation on functions that change their behavior with a canceled context; if it does not mention canceled-context
+behavior, its behavior does not change.
+For exact behavior, make sure to read the documentation on functions that you are calling.
+
+As an incomplete summary, these actions will all fail immediately (or when the activity completes, if WaitForCancellation
+is true), and the associated error returns (possibly within a Future) will be a workflow.CanceledError:
```

Comment:
`or when the activity completes, if WaitForCancellation is true`

Is this the default behavior? I tested this before seems like so. Maybe we should remove activity from this summary and make it as a special case.

Reply:
default behavior is false, so it does not wait.

are you thinking to just drop the exception note (or say "read executeactivity for more details"), or put it in its own list since it (and maybe executechildworkflow?  I'm not sure how it completes either) is kinda unique?

Reply:
> default behavior is false, so it does not wait.
Interesting, looks like Java client is different. I tested in 
https://github.com/uber/cadence-java-samples/pull/47
that it will wait for completion. 

also https://github.com/uber/cadence-java-samples/pull/48 for long running activity activity will get canceled right way when reporting heartbeat. 

Yeah I believe pulling activity out is better to read. if having a separate paragraph.  It has different cases than other. Using `()` is friendly to understand.